models/Invitation.php

```diff
@@ -0,0 +1,212 @@
+            [['created_by'], 'exist', 'skipOnError' => true, 'targetClass' => User::class, 'targetAttribute' => ['created_by' => 'id']],
+            [['user_id'], 'exist', 'skipOnError' => true, 'targetClass' => User::class, 'targetAttribute' => ['user_id' => 'id']],
+        ];
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function attributeLabels()
+    {
+        return [
+            'id' => Yii::t('app', 'ID'),
+            'user_id' => Yii::t('app', 'User ID'),
+            'email' => Yii::t('app', 'Email'),
+            'status' => Yii::t('app', 'Status'),
+            'token' => Yii::t('app', 'Token'),
+            'created_at' => Yii::t('app', 'Created At'),
+            'created_by' => Yii::t('app', 'Created By'),
+        ];
+    }
+
+    /**
+     * Return fields for frontend
+     *
+     * @return array
+     */
+    public function fields()
+    {
+        return array_merge(parent::fields(), [
+            'created_at' => function () {
+                return $this->created_at * 1000;
+            },
+            'use_date' => function () {
+                return $this->use_date ? Yii::$app->formatter->asDatetime($this->use_date) : null;
+            },
+            'statusLabel' => function () {
+                return $this->getStatusLabel();
+            }
+        ]);
+    }
+
+    /**
+     * Extra fields with relation
+     *
+     * @return array|string[]
+     */
+    public function extraFields()
+    {
+        return ['createdBy', 'user'];
+    }
+
+    /**
+     * Gets query for [[CreatedBy]].
+     *
+     * @return ActiveQuery
+     */
+    public function getCreatedBy()
+    {
+        return $this->hasOne(User::class, ['id' => 'created_by']);
+    }
+
+    /**
+     * Gets query for [[User]].
+     *
+     * @return ActiveQuery
+     */
+    public function getUser()
+    {
+        return $this->hasOne(User::class, ['id' => 'user_id']);
+    }
+
+    /**
+     * @param bool $insert
+     * @return bool
+     * @throws Exception
+     */
+    public function beforeSave($insert)
+    {
+        if ($insert) {
+            $this->status = self::STATUS_PENDING;
+            $this->token = Yii::$app->security->generateRandomString(256);
+            $this->expire_date = time() + self::TOKEN_LIFETIME;
+        }
+
+        return parent::beforeSave($insert);
+    }
+
+    /**
+     * After save send invitation email to user
+     *
+     * @param bool $insert
+     * @param array $changedAttributes
+     */
+    public function afterSave($insert, $changedAttributes)
+    {
+        parent::afterSave($insert, $changedAttributes);
+
+        if ($insert) {
+            MailHelper::invitation($this);
+        }
+    }
+
+    /**
+     * Find invitation by valid token
+     *
+     * @param $token
+     * @return array|ActiveRecord|null
+     */
+    public static function findByToken($token)
+    {
+        return static::find()
+            ->andWhere(['token' => $token, 'use_date' => null])
```

Comment:
@skavt It is better to create helper methods in InvitationQuery model like: `byToken`, `notUsed`. So the usage will be like this: `static::find()->byToken($token)->notUsed()`